### PR TITLE
Add support for gzipped erlang binaries in HTTP API

### LINKED ISSUE #318: Riak tries to decode erlang binary when gzipped
When using `Content-Type: application/x-erlang-binary`, Riak calls `binary_to_term` on a PUT body, even if you've set `Content-Encoding: gzip`.

Steps to repro:

In an erlang shell:

``` erlang
T = {some, <<"term">>}.
file:write_file("erlang-binary", term_to_binary(T)).
```

``` shell
gzip erlang-binary

curl -X PUT -H "Content-Encoding: gzip" -H "Content-Type: application/x-erlang-binary" localhost:8091/riak/gzip/test --data-binary @erlang-binary.gz
```

Gives a 500 error, the problem appears to be [here](https://github.com/basho/riak_kv/blob/1.1.2/src/riak_kv_wm_utils.erl#L213-L214).

### PR DESCRIPTION
**Pull request by bsparrow435**
This fixes #318 (RIAK-2366) and adds support for gzipped erlang binaries.

### CODE REVIEW

File: src/riak_kv_wm_utils.erl [hunk 1]
```diff
@@ -226,12 +226,18 @@ encode_value(V) when is_binary(V) ->
...
     term_to_binary(V).
 
--spec accept_value(string(), binary()) -> term().
+-spec accept_value(string(), riak_kv_wm_utils_dict(), binary()) -> term().
 %% @doc Accept the object value as a binary - content type can be used
 %%      to decode
-accept_value("application/x-erlang-binary",V) ->
-    binary_to_term(V);
-accept_value(_Ctype, V) ->
+accept_value("application/x-erlang-binary",MD, V) ->
```
Comment (lukebakken) [inline src/riak_kv_wm_utils.erl:232]: Does this interact at all with basho/riak-erlang-client#60?

Comment (lukebakken) [reply src/riak_kv_wm_utils.erl:232]: Also, should these MIME and content types be a constant / macro? Not sure if we have something in place for that.

Comment (bsparrow435) [reply src/riak_kv_wm_utils.erl:232]: I was testing this incorrectly and have discovered some issues. I'll be updating the PR shortly with some changes.

**Comment by bsparrow435** [comment]
OK, this is confirmed to work with the following test:

```
T = {some, <<"term">>}.
file:write_file("erlang-binary", term_to_binary(T)).
```

```
gzip erlang-binary
curl -X PUT -H "Content-Encoding: gzip" -H "Content-Type: application/x-erlang-binary" 127.0.0.1:10018/types/test/buckets/test/keys/gzip --data-binary @erlang-binary.gz
```

**Comment by bsparrow435** [comment]
@lukebakken I absolutely agree that the MIME types should be encapsulated in a macro. With that said, i was talking with Torben and others earlier about cleaning up all the records and macros in webmachine/kv(they are an absolute mess). I believe encapsulating these types would be included into that general solution.

**Pull request merged**